source/unified-test-format/schema-1.18.json

```diff
@@ -432,6 +428,42 @@
                         }
                     }
                 }
+            },
+            "type": "object",
+            "additionalProperties": false,
+            "patternProperties": {
+                "^aws:[a-zA-Z0-9_]+?$": {
+                    "$ref": "#/definitions/kmsProviders/$defs/AWSKMSOptions"
```

Comment:
Since the named providers don't support placeholders, I think this may warrant different definitions. You can probably revert to inlined definitions for "aws", "gcp" and "azure".

You may not even need `$defs` for the "kmip" and "local" providers. If the "kmip" and "local" providers are identical, you can just tweak just their regex patterns to make the entire suffix optional (e.g. `^local(:[a-zA-Z0-9_]+)?$`) so it will apply to both cases.

Reply:
> Since the named providers don't support placeholders, I think this may warrant different definitions. You can probably revert to inlined definitions for "aws", "gcp" and "azure".

I think the named and unnamed can use the same definition. The `aws:name2` KMS provider uses placeholders, since it is expected to be filled in with credentials for a different AWS account.

> you can just tweak just their regex patterns to make the entire suffix optional (e.g. `^local(:[a-zA-Z0-9_]+)?$`) so it will apply to both cases.

I like that idea. Since the named providers also support placeholders, the named and unnamed properties have been combined.

Reply:
> Since the named providers also support placeholders, the named and unnamed properties have been combined

Ah, I mixed up support for placeholders with automatic credentials. The automatic credentials are what the named providers don't support.